### PR TITLE
refactor: inject pwa client plugin from module

### PR DESCRIPTION
**Pull request by danielroe**

### CODE REVIEW

**Comment by userquin** [comment]
We should protect Pwa** sfc, we're using them directly and so it will fail on dev, check `PwaPrompt` usage on default layout

**Comment by danielroe** [comment]
I think they're all protected with `v-if`...

**Comment by userquin** [comment]
you're right, `$pwa?.`

File: pages/share-target.vue [hunk 1]
```diff
@@ -8,7 +8,7 @@ definePageMeta({
 
 useWebShareTarget()
 
-const pwaIsInstalled = process.server ? false : useNuxtApp().$pwa.isInstalled
+const pwaIsInstalled = process.client && !!useNuxtApp().$pwa?.isInstalled
```
Comment (userquin) [inline pages/share-target.vue:11]: the middleware runs before? if so, you can replace it with `useNuxtApp().$pwa.isInstalled`

**Review by userquin — APPROVED** [review]

**Pull request merged**